tools/adjust_rc_with_feedback.py

```diff
@@ -165,60 +165,71 @@ def update_rating_curve(fim_directory, water_edge_median_ds, htable_path, output
 
-            # Create the adjust_ManningN column by combining the hydroid_ManningN with the featid_ManningN (use feature_id value if the hydroid is in a feature_id that contains valid hydroid_ManningN value(s))
+            ## Create the adjust_ManningN column by combining the hydroid_ManningN with the featid_ManningN (use feature_id value if the hydroid is in a feature_id that contains valid hydroid_ManningN value(s))
             conditions  = [ (df_nmerge['hydroid_ManningN'].isnull()) & (df_nmerge['featid_ManningN'].notnull()), (df_nmerge['hydroid_ManningN'].isnull()) & (df_nmerge['featid_ManningN'].isnull()) & (df_nmerge['group_ManningN'].notnull()) ]
             choices     = [ df_nmerge['featid_ManningN'], df_nmerge['group_ManningN'] ]
             df_nmerge['adjust_ManningN'] = np.select(conditions, choices, default=df_nmerge['hydroid_ManningN'])
             df_nmerge.drop(['feature_id','NextDownID','LENGTHKM','LakeID','order_'], axis=1, inplace=True) # drop these columns to avoid duplicates where merging with the full hydroTable df
             
-            # Optional ouputs: 1) merge_n_csv csv with all of the calculated n values and 2) a catchments .gpkg with new joined attributes
-            if optional_outputs == 'True':
+            ## Update the catchments polygon .gpkg with joined attribute - "src_calibrated"
+            if os.path.isfile(catchments_poly_path):
+                input_catchments = gpd.read_file(catchments_poly_path)
+                ## Create new "src_calibrated" column for viz query
+                if 'src_calibrated' in input_catchments.columns: # check if this attribute already exists and drop if needed
+                    input_catchments.drop(['src_calibrated'], axis=1, inplace=True)
+                df_nmerge['src_calibrated'] = np.where(df_nmerge['adjust_ManningN'].notnull(), 'True', 'False')
+                output_catchments = input_catchments.merge(df_nmerge[['HydroID','src_calibrated']], how='left', on='HydroID')
+                output_catchments['src_calibrated'].fillna('False', inplace=True)
+                output_catchments.to_file(catchments_poly_path,driver="GPKG",index=False) # overwrite the previous layer
+                df_nmerge.drop(['src_calibrated'], axis=1, inplace=True)
+            ## Optional ouputs: 1) merge_n_csv csv with all of the calculated n values and 2) a catchments .gpkg with new joined attributes
+            if optional_outputs == 'True' and os.path.isfile(catchments_poly_path):
                 output_merge_n_csv = os.path.join(fim_directory, huc, 'merge_src_n_vals_' + huc + '.csv')
                 df_nmerge.to_csv(output_merge_n_csv,index=False)
-                # output new catchments polygon layer with the new manning's n value attributes appended
-                if catchments_poly_path != '':
+                ## output new catchments polygon layer with several new attributes appended
+                if os.path.isfile(catchments_poly_path):
```

Comment:
if statement might be redundant due to same check added on line 201

Reply:
Good find. I have addressed this.